devito/xdsl_core/xdsl_cpu.py

```diff
@@ -652,9 +658,26 @@ def generate_tiling_arg(nb_tiled_dims: int):
     """
     if nb_tiled_dims < 1:
         return 'parallel-loop-tile-sizes=0'
+
+    # TOFIX: 64 is hardcoded, should be a parameter
+    # TOFIX: Zero is also hardcoded, should be a parameter
     return "parallel-loop-tile-sizes=" + ",".join(["64"]*nb_tiled_dims) + ",0"
 
 
+def generate_collapse_arg(num_loops: int):
+    """
+    Generate the number of loops that will be collapsed
+    Resort to 1 if no number of loops is provided
+    """
+
+    if num_loops < 1:
+        num_loops = 1
+
+    ret_arg = "collapse=" + "".join(str(num_loops))  # noqa
+
+    return ret_arg
+
+
```

Comment:
More a thought than a request :wink: 
It might be simpler to pass kwargs here and encapsulate the whole convert-scf-to-openmp pass? To later play with schedules, and other omp knobs like inner omp-vectorized loops and such

Reply:
will leave this for future fixes...